### ISSUE TITLE
Points don't inherit font family

### ISSUE THREAD

**Issue by botanize** [tidyverse/ggplot2#4305]
I expect that the font-family specified in the theme is applied to all applicable graphical content. However, when specifying some unicode characters for the shape scale I get empty boxes instead of the characters. Point geometries can use arbitrary characters for the shape scale, but not all characters are available in all font families. While constructing the points grob, `GeomPoint` omits the `fontfamily` `gpar` parameter, which means that there is no way to specify a font family that contains the desired characters. The standard ways of specifying a font family for the quartz device don't work for the gpar fontfamily (`quartz(family =`, or  `quartzFonts(sans = quartzFont(…`).

``` r
library(ggplot2)
library(grid)

theme_set(theme_minimal(base_family = 'Apple SD Gothic Neo'))

p = ggplot(data.frame(x = 1:3, y = 1:3, shape = factor(0:2)), aes(x, y)) + geom_point(aes(shape = shape), size = 4) + scale_shape_manual(values = c('0' = '\u2013', '1' = '◀', '2' = '\u25B6')) + theme(text = element_text(family = 'Apple SD Gothic Neo'))
print(p)
data = ggplot_build(p)
gtable = ggplot_gtable(data)
grid.draw(gtable)
```

![](https://i.imgur.com/lv0ezb5.png)

``` r
# add family to gpar for points grob
gtable$grobs[[6]]$children[[3]]$gp$fontfamily = 'Apple SD Gothic Neo'
grid.newpage()
grid.draw(gtable)
```

![](https://i.imgur.com/O63iuIq.png)

<sup>Created on 2020-12-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I get the behavior I want by adding the font family `gpar` parameter to the correct grobs as above or to the gtable itself:
```r
data = ggplot_build(p)
gtable = ggplot_gtable(data)
gtable$gp = gpar(fontfamily = 'Apple SD Gothic Neo')
grid.draw(gtable)
```

From a user perspective, I think this could be fixed by either adding a family parameter to `scale_shape` or inheriting the family from the theme or device.

**Comment by thomasp85** [comment]
I agree about your request for this functionality - in the meantime you can use the ragg devices to avoid the boxes as it should be able to figure out a fallback font

**Comment by teunbrand** [comment]
The layers currently don't know about the theme. Wouldn't it be simpler to add a `font` argument to `geom_point()`?

**Comment by teunbrand** [comment]
After some consideration, I've come to argue that we shouldn't fix this.
The reason is that `geom_point()` is/should be for shapes 0-25, not for text.
To [quote Hiroaki](https://github.com/tidyverse/ggplot2/pull/5143#issuecomment-1439249120):

> I think ggplot2 should have prohibited any use of these single characters as a shape. We can use arbitrary letter here, but I feel this isn't a "point" geom anymore.

The correct solution is to use a text geom, with a label scale. If the argument is 'but then we cannot use the `shape` symbols', you'd be right, but there are plenty of unicode shapes that capture all types of symbols you can use. As a last resort, you can divide the data over a point/text layer.

Unfortunately, `manual_scale()` is not exported, but that is a matter for a different issue. We can easily redirect e.g. `scale_colour_manual()` instead.

``` r
library(ggplot2)

data.frame(x = 1:3, y = 1:3, shape = factor(0:2)) |>
  ggplot(aes(x, y)) + 
  geom_text(aes(label = shape), size = 4, family = "Montserrat") +
  scale_colour_manual(
    values = c('0' = '\u2013', '1' = '◀', '2' = '\u25B6'), 
    aesthetics = "label"
  )
```

![](https://i.imgur.com/jv2sysK.png)

<sup>Created on 2025-02-24 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
